prometheus/testutil/testutil.go

```diff
@@ -112,31 +112,43 @@ func ToFloat64(c prometheus.Collector) float64 {
 	panic(fmt.Errorf("collected a non-gauge/counter/untyped metric: %s", pb))
 }
 
-// CollectAndCount collects all Metrics from the provided Collector and returns their number.
-//
-// This can be used to assert the number of metrics collected by a given collector after certain operations.
-//
-// This function is only for testing purposes, and even for testing, other approaches
-// are often more appropriate (see this package's documentation).
-func CollectAndCount(c prometheus.Collector) int {
-	var (
-		mCount int
-		mChan  = make(chan prometheus.Metric)
-		done   = make(chan struct{})
-	)
-
-	go func() {
-		for range mChan {
-			mCount++
-		}
-		close(done)
-	}()
+// CollectAndCount registers the provided Collector with a newly created
+// pedantic Registry. It then calls GatherAndCount with that Registry and with
+// the provided metricNames. In the unlikely case that the registration or the
+// gathering fails, this function panics. (This is inconsistent with the other
+// CollectAnd… functions in this package and has historical reasons. Changing
```

Comment:
Do we really need to have error? Previous version was error-free

Reply:
And the current version is still error-free.

Reply:
Sure, but you plan to make it error full (: